Unify logging by using logrus for jetstream logs

dvob opened the pull request:
I guess tests for the logging is rather unusual so I omitted tests for this change.

* [x] I have added Go unit tests or [Complement integration tests](https://github.com/matrix-org/complement) for this PR _or_ I have justified why this PR doesn't need tests
* [x] Pull request includes a [sign off below using a legally identifiable name](https://matrix-org.github.io/dendrite/development/contributing#sign-off) _or_ I have already signed off privately

Signed-off-by: `David Schneider <dsbrng25b@gmail.com>`

S7evinK commented on setup/jetstream/nats.go:
```diff
@@ -53,7 +53,7 @@ func (s *NATSInstance) Prepare(process *process.ProcessContext, cfg *config.JetS
 			panic(err)
 		}
-		s.ConfigureLogger()
+		s.SetLogger(NewLogAdapter(), false, false)
```
Maybe worth to explain that `false, false` is `debugFlag` and `traceFlag`

dvob replied:
Changed this to use the values from the NATS options which mimics the exact behavior of ConfigureLogger which was used before.

S7evinK commented on setup/jetstream/log.go:
```diff
@@ -0,0 +1,42 @@
+}
+
+func (l *LogAdapter) Warnf(format string, v ...interface{}) {
+	l.entry.Errorf(format, v...)
```
Looks like there are a few copy and paste mistakes? e.g. `Warnf` logging on `Errorf` and `Fatalf` only on `Errorf`?

dvob commented:
@S7evinK Thank you for the review. 

Yes these there were some copy/paste mistakes which I fixed.

I also changed the call to SetLogger to use the values from the NATS options. This should make the call more clear and behave exactly as ConfigureLogger would.

S7evinK reviewed (approved):
Looks good! Thanks!
